table-api/src/main/java/io/deephaven/api/updateby/UpdateByControl.java

```diff
@@ -26,32 +31,88 @@ public static UpdateByControl defaultInstance() {
         return builder().build();
     }
 
+    public static boolean useRedirectionDefault() {
+        return Boolean.getBoolean("UpdateByControl.useRedirection");
+    }
+
+    public static int chunkCapacityDefault() {
+        return Integer.getInteger("UpdateByControl.chunkCapacity", 4096);
+    }
+
+    public static double maximumStaticMemoryOverheadDefault() {
+        return Double.parseDouble(System.getProperty("UpdateByControl.maximumStaticMemoryOverhead", "1.1"));
+    }
+
+    public static int initialHashTableSizeDefault() {
+        return Integer.getInteger("UpdateByControl.initialHashTableSize", 4096);
+    }
+
+    public static double maximumLoadFactorDefault() {
+        return Double.parseDouble(System.getProperty("UpdateByControl.maximumLoadFactor", "0.75"));
+    }
+
+    public static double targetLoadFactorDefault() {
+        return Double.parseDouble(System.getProperty("UpdateByControl.targetLoadFactor", "0.7"));
+    }
+
+    public static MathContext mathContextDefault() {
+        final String p = System.getProperty("UpdateByControl.mathContext", "DECIMAL64");
+        switch (p) {
+            case "UNLIMITED":
+                return MathContext.UNLIMITED;
+            case "DECIMAL32":
+                return MathContext.DECIMAL32;
+            case "DECIMAL64":
+                return MathContext.DECIMAL64;
+            case "DECIMAL128":
+                return MathContext.DECIMAL128;
+            default:
+                throw new IllegalArgumentException("Unexpected UpdateByControl.mathContext: " + p);
+        }
+    }
+
+    @Nullable
+    public abstract Boolean useRedirection();
+
+    public abstract OptionalInt chunkCapacity();
+
+    public abstract OptionalDouble maxStaticSparseMemoryOverhead();
+
+    public abstract OptionalInt initialHashTableSize();
+
+    public abstract OptionalDouble maximumLoadFactor();
+
+    public abstract OptionalDouble targetLoadFactor();
+
+    public abstract Optional<MathContext> mathContext();
+
     /**
-     * if redirections should be used for output sources instead of sparse array sources.
+     * If redirections should be used for output sources instead of sparse array sources.
      *
      * <p>
      * Default is `false`. Can be changed with system property {@code UpdateByControl.useRedirection} or by providing a
```

Comment:
Should we move these comments (esp re: default values) to the functions where the system properties are read (vs. the `OrDefault()` versions? Probably OK either way but might get overlooked when/if defaults are changed.